NixSupport/devenv.nix

```diff
@@ -0,0 +1,86 @@
+{ pkgs, config, lib, ... }:
+
+let
+  cfg = config.ihp;
+  types = lib.types;
+in
+{
+  options.ihp = {
+    enable = lib.mkEnableOption "Enable IHP support";
+
+    ghcCompiler = lib.mkOption {
+      default = pkgs.haskell.packages.ghc944;
+
+      description = ''
+        The GHC compiler to use for IHP.
+      '';
+    };
+
+    haskellPackages = lib.mkOption {
+      default = p: with p; [
+        cabal-install
+        base
+        wai
+        text
+        hlint
+        ihp
+      ];
+      description = ''
+        List of Haskell packages to be installed in the IHP environment.
+      '';
+    };
+
+    projectPath = lib.mkOption {
+      type = types.path;
+      description = ''
+        Path to the IHP project.
+      '';
+    };
+  };
+
+  config =
+    let
+      ghcCompiler = import ./mkGhcCompiler.nix {
+        inherit pkgs;
+        inherit (cfg) ghcCompiler;
+        ihp = ../.;
+      };
+    in
+    lib.mkIf cfg.enable {
+      packages = [ ghcCompiler.ihp ]; 
+
+      languages.haskell.enable = true;
+      languages.haskell.package = ghcCompiler.ghc.withPackages cfg.haskellPackages;
+
+      scripts.start.exec = ''
+        ${ghcCompiler.ihp}/bin/RunDevServer
+      '';
+
+      processes.devServer.exec = "start";
+
+      env.IHP_DEVENV = "1";
+      env.DATABASE_URL = "postgres:///app?host=${config.env.PGHOST}";
+
+      services.postgres.enable = true;
+      services.postgres.initialDatabases = [
+        {
+          name = "app";
+          schema = pkgs.runCommand "ihp-schema" {} ''
+            touch $out
+
+            echo "-- IHPSchema.sql" >> $out
+            echo "" >> $out
+            cat ${../lib/IHP/IHPSchema.sql} | sed -e s'/--.*//' | sed -e s'/$/\\/' >> $out
```

Comment:
The postgres module makes use of `postgres` single user mode to apply the initial schema which has some quirks with regard to statement terminators.

Basically it does not support multiline statements unless the `-j` option is used, which is unfortunately not the case with devenv.

So in order to workaround this we remove all comments from the file and add `\` to each line so that postgres sees each file as a single statement with multiple commands.

I will open a PR to improve the devenv module because it clearly isn't working as expected.

Further reading : https://www.postgresql.org/docs/current/app-postgres.html#APP-POSTGRES-SINGLE-USER

Reply:
While testing I ran into an interesting problem: The SQL command that are executed are written to the postgres logs. On a project with a large `Fixtures.sql` (~8mb, 30k inserts), this will keep devenv running for minutes trying to print every SQL statement from the file.

Not sure whether we can fix this here or we need to make some changes to the devenv upstream to fix this

Reply:
I think this is going to be the main issue left with this PR because it depends on the upstream package.  Either we fix this by still loading of the initial schema ourselves with the `IHP.IDE.Postrges` module or improve how devenv's postgres module loads the initial databases schemas.

I don't have a strong opinion on either ways, so let me know what you think about this. 🙏

Reply:
Let's make an issue for the devenv repo and get this improved upstream 👍  Most projects have small fixtures, so it's only an issue in edge cases and can be fixed in the future